src/Cli/dotnet/commands/dotnet-workload/install/WorkloadInstallManager.cs

```diff
@@ -0,0 +1,100 @@
+// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.Deployment.DotNet.Releases;
+using Microsoft.DotNet.Cli;
+using Microsoft.DotNet.Cli.Utils;
+using Microsoft.NET.Sdk.WorkloadManifestReader;
+using Product = Microsoft.DotNet.Cli.Utils.Product;
+
+namespace Microsoft.DotNet.Workloads.Workload.Install
+{
+    internal class WorkloadInstallManager
+    {
+        private readonly IReporter _reporter;
+        private readonly IWorkloadInstaller _workloadInstaller;
+        private readonly IWorkloadResolver _workloadResolver;
+        private readonly ReleaseVersion _sdkVersion;
+
+        public WorkloadInstallManager(
+            IReporter reporter,
+            IWorkloadInstaller workloadInstaller,
+            IWorkloadResolver workloadResolver,
+            string version = null)
+        {
+            _sdkVersion = new ReleaseVersion(version ?? Product.Version);
+            _workloadInstaller = workloadInstaller;
+            _workloadResolver = workloadResolver;
+            _reporter = reporter;
+        }
+
+        public void InstallWorkloads(IEnumerable<WorkloadId> workloadIds, bool skipManifestUpdate = false)
+        {
+            _reporter.WriteLine();
+            var featureBand = new SdkFeatureBand(string.Join('.', _sdkVersion.Major, _sdkVersion.Minor, _sdkVersion.SdkFeatureBand));
+
+            if (!skipManifestUpdate)
+            {
+                throw new NotImplementedException();
+            }
+
+            InstallWorkloadsWithInstallRecord(workloadIds, featureBand);
+
+            if (_workloadInstaller.GetInstallationUnit().Equals(InstallationUnit.Packs))
+            {
+                (_workloadInstaller as PackWorkloadInstallerBase).GarbageCollectInstalledWorkloadPacks();
+            }
+
+            _reporter.WriteLine();
+            _reporter.WriteLine(string.Format(LocalizableStrings.InstallationSucceeded, string.Join(", ", workloadIds)));
```

Comment:
I'm not sure if joining items with commas is localization-neutral.  I'm not sure if there's a better pattern.

Reply:
It is good enough. Maybe join by whitespace. But so far that's how existing code handle it.